CVEProject/cve-services
# Resolves issue #609, Added validators and sanitizers for role creation and update

Closes Issue #609 

# Summary
Added additional data validation and sanitization for role fields in POST and PUT requests

# Important Changes
`src/middleware/middleware.js`
- Added validator `isFlatStringArray()` and sanitizer `toUpperCaseArray()`

`src/controller/org.controller/index.js`
- Refactored controllers to use new validator and sanitizer
- Code cleanup

# Testing

Steps to manually test updated functionality, if possible:

`POST /org/:shortname/user` - Confirm each of the follow values for the `active_roles` field fails with a 400 status, valid JSON response, and error `Parameter must be a one-dimensional array of strings`
- [x] `{ "a": "ADMIN" }`
- [x] `[{ "a": "ADMIN" }]`

`PUT /org/:shortname/user` - Confirm the following query strings fail with the same error as above:
- [x] `active_roles.add[][a]=CNA`
- [x] `active_roles.add[][CNA]`

`POST /org` - Repeat the same process as the POST create user endpoint

`PUT /org/:shortname` - Repeat the same process as the PUT update user endpoint


# Notes
- HTTP tests are included.
